x-pack/plugins/cases/server/services/transform.ts

```diff
@@ -95,3 +101,35 @@ export function transformFieldsToESModel(connector: CaseConnector): ConnectorPer
     []
   );
 }
+
+export function transformCustomFieldsToESModel(customFields: CaseCustomFields) {
+  return customFields.map((customField) => {
+    if (customField.type === CustomFieldTypes.LIST && customField.value) {
+      const [selectedKey] = Object.keys(customField.value);
+      const [value] = Object.values(customField.value);
+      return {
+        type: CustomFieldTypes.LIST,
+        key: `${customField.key}.${selectedKey}`,
+        value,
+      };
+    }
+    return customField;
+  });
+}
+
+export function transformCustomFieldsToExternalModel(
+  customFields: NonNullable<CasePersistedAttributes['customFields']>
+) {
+  return customFields.map((customField) => {
+    if (customField.type === CustomFieldTypes.LIST && customField.value) {
+      const [key, selectedKey] = customField.key.split('.');
```

Comment:
Users can set the configuration via API and looking at `ConfigurationPatchRequestRt` I think they can set their own keys(I didn't test, just looked at the code. Please correct me if I am wrong 😅 ).

This opens the possibility of keys with `.`s in them in which case, this logic will break.

Reply:
Yeah that's potentially an issue with the schema in the acceptance requirements for this feature. Not really sure what to do about this.

Reply:
Actually maybe we can handle this by making `selectedKey` the last value in the `.split` array and `key` we just `join('.')` everything else.